src/main/java/org/graylog/collector/file/PathSet.java

```diff
@@ -0,0 +1,195 @@
+import java.nio.file.FileVisitor;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.List;
+import java.util.Set;
+
+public class PathSet {
+    private static final Logger LOG = LoggerFactory.getLogger(PathSet.class);
+
+    // List taken from sun.nio.fs.Globs.
+    private static final List<String> GLOB_META_CHARS = ImmutableList.of("\\", "*", "?", "[", "{");
+
+    private final String pattern;
+    private final FileTreeWalker fileTreeWalker;
+    private final PathMatcher matcher;
+    private final Path rootPath;
+
+    public static class GlobbingFileVisitor extends SimpleFileVisitor<Path> {
+        private final PathMatcher matcher;
+        private final ImmutableSet.Builder<Path> matchedPaths;
+
+        public GlobbingFileVisitor(PathMatcher matcher, ImmutableSet.Builder<Path> matchedPaths) {
+            this.matcher = matcher;
+            this.matchedPaths = matchedPaths;
+        }
+
+        @Override
+        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+            // Skip /proc because it can throw some permission errors we cannot check for.
+            return dir.startsWith("/proc") ? FileVisitResult.SKIP_SUBTREE : FileVisitResult.CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+            LOG.warn("Unable to change into directory {} - Check permissions", file);
+            return FileVisitResult.SKIP_SUBTREE;
+        }
+
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attributes) throws IOException {
+            if (matcher.matches(file)) {
+                // TODO needs to be an absolute path because otherwise the FileObserver does weird things. Investigate what's wrong with it.
+                matchedPaths.add(file.toAbsolutePath());
+            }
+
+            return FileVisitResult.CONTINUE;
+        }
+    }
+
+    public interface FileTreeWalker {
+        void walk(Path basePath, FileVisitor<Path> visitor) throws IOException;
+    }
+
+    /**
+     * Default FileTrackingList that uses the file system to scan for files to track.
+     *
+     * @param pattern the pattern to scan for
+     */
+    public PathSet(final String pattern) {
+        this(pattern, new FileTreeWalker() {
+            @Override
+            public void walk(Path basePath, FileVisitor<Path> visitor) throws IOException {
+                Files.walkFileTree(basePath, visitor);
+            }
+        });
+    }
+
+    public PathSet(final String pattern, FileTreeWalker fileTreeWalker) {
+        this.pattern = pattern;
+        this.fileTreeWalker = fileTreeWalker;
+        final String patternString = Utils.isWindows() ? pattern.replace("\\", "\\\\") : pattern;
```

Comment:
No, does not work. Using `Pattern.quote("C:\test\*.log")` creates a regex like this: `^QC:test\*\.logE$` That's not going to work.

The globbing code does something special...